### PR TITLE
Validate cell syntax

### PR DESCRIPTION
**Pull request by xbrianh**

### CODE REVIEW

File: herzog/parser.py [hunk 1]
```diff
@@ -82,6 +82,16 @@ def __iter__(self):
     def rewind(self):
         self._rewind = True
 
+def _validate_cell(cell_lines: List[str], line_number: Optional[int]=None):
+    line_number_str = str(line_number) if line_number is not None else "?"
+    if not cell_lines:
+        raise SyntaxError(f"line {line_number_str}: Expected Herzog cell content")
```
Comment (DailyDreaming) [inline herzog/parser.py:88]: Nit: Raise this first?  Consider:

```
def _validate_cell(cell_lines: List[str], line_number: Optional[int]=None):
    if not cell_lines:
        raise SyntaxError(f"line {line_number_str}: Expected Herzog cell content")

    line_number_str = str(line_number) if line_number is not None else "?"
    try:
        compile(os.linesep.join(cell_lines), "", "exec")
    except SyntaxError:
        raise SyntaxError(f"line {line_number_str}")
```

**Review by DailyDreaming — APPROVED** [review]
LGTM.  See comment.

File: herzog/parser.py (hunk 1 above)
Comment (xbrianh) [reply herzog/parser.py:88]: The error messages require `line_number_str` to be defined first.

Comment (DailyDreaming) [reply herzog/parser.py:88]: Ah, missed that, my bad.

**Pull request merged**